src/herder/HerderSCPDriver.cpp

```diff
@@ -687,59 +691,34 @@ HerderSCPDriver::combineCandidates(uint64_t slotIndex,
     }
 
     // take the txSet with the biggest size, highest xored hash that we have
-    TxSetFramePtr bestTxSet;
+    using HighestPair = std::pair<StellarValue, TxSetFramePtr>;
+    HighestPair highestPair = std::accumulate(
+        candidateValues.begin(), candidateValues.end(), HighestPair(),
+        [&](HighestPair const& hp, StellarValue const& sv) {
+            auto const& highestTxSet = hp.second;
+            auto const& svSetHash = sv.txSetHash;
+            auto const cTxSet = mPendingEnvelopes.getTxSet(svSetHash);
+            return (cTxSet && cTxSet->previousLedgerHash() == lcl.hash &&
+                    (!highestTxSet ||
+                     compareTxSets(highestTxSet, cTxSet, hp.first.txSetHash,
+                                   svSetHash, lcl.header, candidatesHash)))
+                       ? HighestPair(sv, cTxSet)
+                       : hp;
+        });
+
+    comp = highestPair.first;
+    if (!curProtocolPreservesTxSetCloseTimeAffinity())
     {
-        Hash highest;
-        TxSetFramePtr highestTxSet;
-        for (auto const& sv : candidateValues)
-        {
-            TxSetFramePtr cTxSet = mPendingEnvelopes.getTxSet(sv.txSetHash);
-
-            if (cTxSet && cTxSet->previousLedgerHash() == lcl.hash)
-            {
-                if (compareTxSets(highestTxSet, cTxSet, highest, sv.txSetHash,
-                                  lcl.header, candidatesHash))
-                {
-                    highestTxSet = cTxSet;
-                    highest = sv.txSetHash;
-                }
-            }
-        }
-        // make a copy as we're about to modify it and we don't want to mess
-        // with the txSet cache
-        bestTxSet = std::make_shared<TxSetFrame>(*highestTxSet);
```

Comment:
Correct -- the "modify it" that the old comment referred to was the `trimInvalid()` section, which I've removed as [noted above](https://github.com/stellar/stellar-core/pull/2625#discussion_r461229945).

It occurs to me in light of your question, though, that I could fling a bunch of `const`s around to make it clear (and compiler-checked) that we're _not_ modifying anything (and therefore don't need to copy anything).  I've done so in an additional commit (554b79ff4059442b8d2de3fb4128c2f91041e10b) just in case anyone thinks that that's not a good idea, in which case I could easily remove it again.